### DISCUSSION TITLE
Setting bearer token in Vue Apollo v4?

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [Akryum/vue-apollo#1327] [Q&A]
Hello everyone, when googling, to set headers I would use something like this

```
const authLink = setContext(async (_, { headers }) => {
  // get the authentication token from local storage if it exists
  const token = JSON.parse(localStorage.getItem('apollo-token'))
  // Return the headers to the context so httpLink can read them
  return {
    headers: {
      ...headers,
      authorization: token || ''
    }
  }
})
```

But `setContext` seems to have been removed, and `uri` parameter has moved from the client to the link, so this approach doesn't work. Any way to replicated this behaviour?

**Comment by dominikklein** [comment]
I think this more a apollo client question and here the setContext should still exists. For example `import { setContext } from '@apollo/client/link/context'`.
I'm doing something like this with the current apollo client + vue apollo v4:

```
const setAuthorizationLink = setContext((request, { headers }) => ({
  headers: {
    ...headers,
    'X-CSRF-Token': getCSRFToken(),
  },
}))
```
